sessionctx/variable/sysvar_test.go

```diff
@@ -845,6 +845,18 @@ func TestDefaultCharsetAndCollation(t *testing.T) {
 	require.Equal(t, val, mysql.DefaultCollationName)
 }
 
+func TestInstanceScope(t *testing.T) {
+
+	// Instance scope used to be settable via "SET SESSION", which is weird to any MySQL user.
+	// It is now settable via SET GLOBAL, but to work correctly a sysvar can only ever
+	// be INSTANCE scoped or GLOBAL scoped, never *both* at the same time (at least for now).
+	// Otherwise the semantics are confusing to users for how precedence applies.
+
+	for _, sv := range GetSysVars() {
+		require.False(t, sv.HasGlobalScope() && sv.HasInstanceScope(), "sysvar %s has both instance and global scope", sv.Name)
+	}
+}
+
```

Comment:
Please add some really simple tests using the internal API. Like in this function: https://github.com/pingcap/tidb/blob/7bf5e4e23ea05cabe360d57f9963ec6b78c414e6/sessionctx/variable/sysvar_test.go#L81-L113